test/functional/rpc_getorphantxs.py

```diff
@@ -0,0 +1,122 @@
+        peer.send_and_ping(msg_tx(tx_child_1["tx"]))
+        peer.send_and_ping(msg_tx(tx_child_2["tx"]))
+
+        self.log.info("Check that neither parent is in the mempool")
+        assert_equal(node.getmempoolinfo()["size"], 0)
+
+        self.log.info("Check that both children are in the orphanage")
+
+        assert_equal(len(node.getorphantxs()), 2)
+        assert node.tx_in_orphanage(tx_child_1["tx"])
+        assert node.tx_in_orphanage(tx_child_2["tx"])
+
+        self.log.info("Broadcast parent 1")
+        peer.send_and_ping(msg_tx(tx_parent_1["tx"]))
+        self.log.info("Check that parent 1 and child 1 are in the mempool")
+        raw_mempool = node.getrawmempool()
+        assert_equal(len(raw_mempool), 2)
+        assert tx_parent_1["txid"] in raw_mempool
+        assert tx_child_1["txid"] in raw_mempool
+
+        self.log.info("Check that orphanage only contains child 2")
+        orphanage = node.getorphantxs()
+        assert_equal(len(orphanage), 1)
+        assert node.tx_in_orphanage(tx_child_2["tx"])
+
+        peer.send_and_ping(msg_tx(tx_parent_2["tx"]))
+        self.log.info("Check that all parents and children are now in the mempool")
+        raw_mempool = node.getrawmempool()
+        assert_equal(len(raw_mempool), 4)
+        assert tx_parent_1["txid"] in raw_mempool
+        assert tx_child_1["txid"] in raw_mempool
+        assert tx_parent_2["txid"] in raw_mempool
+        assert tx_child_2["txid"] in raw_mempool
+        self.log.info("Check that the orphanage is empty")
+        assert_equal(len(node.getorphantxs()), 0)
+
+        self.log.info("Confirm the transactions (clears mempool)")
+        self.generate(node, 1)
+
+    def test_orphan_details(self):
+        self.log.info("Check the transaction details returned from getorphantxs")
+        node = self.nodes[0]
+
+        self.log.info("Create two orphans, from different peers")
+        tx_parent_1 = self.wallet.create_self_transfer()
+        tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
+        tx_parent_2 = self.wallet.create_self_transfer()
+        tx_child_2 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_2["new_utxo"])
+        peer_1 = node.add_p2p_connection(P2PInterface())
+        peer_2 = node.add_p2p_connection(P2PInterface())
+        peer_1.send_and_ping(msg_tx(tx_child_1["tx"]))
+        peer_2.send_and_ping(msg_tx(tx_child_2["tx"]))
+
+        orphanage = node.getorphantxs(2)
+        assert node.tx_in_orphanage(tx_child_1["tx"])
+        assert node.tx_in_orphanage(tx_child_2["tx"])
+
+        self.log.info("Check that orphan 1 and 2 were from different peers")
+        assert orphanage[0]["from"][0] != orphanage[1]["from"][0]
+
+        self.log.info("Unorphan child 2")
+        peer_2.send_and_ping(msg_tx(tx_parent_2["tx"]))
+
+        self.log.info("Checking orphan details (verbosity 1)")
+        orphanage = node.getorphantxs(1)
+        orphan_1 = orphanage[0]
```

Comment:
I was briefly confused about ordering stability, and then realized this has length 1

Reply:
Thanks, yeah it's a bit too implicit so I added `assert_equal(len(node.getorphantxs()), 1)` to make it clearer and more explicit.